lib/object-mask.js

```diff
@@ -396,34 +405,30 @@ function invert(mask) {
 	return result;
 }
 
-// Throws on (deep) attempt to subtract collection from scalar
 function subtract(a, b) {
 	// base cases:
+	if (a === true) return invert(b);
+	if (a === false) return false;
+	if (!a) return invert(b);
 	if (!b) return a;
-	if (!a || b === true || _.isEqual(a, b)) return false;
-	if (objtools.isScalar(a) && !objtools.isScalar(b)) {
-		throw new XError(XError.INVALID_ARGUMENT, {
-			message: 'cannot subtract a non-scalar from a scalar',
-			data: { a: a, b: b }
-		});
-	}
+	if (b === true || _.isEqual(a, b)) return false;
 
-	let key;
+	a = objtools.deepCopy(a);
```

Comment:
I'm not sure why you're doing it, but regardless, it turns an O(n) function into O(n^2).  `subtract()` calls itself recursively as a function of the depth of the parameters.  `deepCopy()` in itself is O(n), being called inside another O(n) recursive loop.

So, yes, I'd like to know why this is here.  More specifically, I'd like you to justify making this function run in polynomial time (and if there isn't a good reason, this function really needs to run in O(n) because it's used in frequently-run code elsewhere).